staging/src/k8s.io/pod-security-admission/metrics/metrics.go

```diff
@@ -16,9 +16,87 @@ limitations under the License.
 
 package metrics
 
+import (
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	Version "k8s.io/component-base/version"
+	"k8s.io/pod-security-admission/api"
+	"strconv"
+)
+
+const (
+	namespace = "pod_security"
+)
+
+var SecurityEvaluation = metrics.NewCounterVec(
+	&metrics.CounterOpts{
+		Name:           "pod_security_evaluations_total",
+		Help:           "Counter of pod security evaluations.",
+		StabilityLevel: metrics.ALPHA,
+	},
+	[]string{"decision", "policy_level", "policy_version", "mode", "operation", "resource", "subresource"},
+)
+
+type Decision string
+type Mode string
+
+const (
+	ModeAudit      = "audit"
+	ModeEnforce    = "enforce"
+	ModeWarn       = "warn"
+	DecisionAllow  = "allow"  // Policy evaluated, request allowed
+	DecisionDeny   = "deny"   // Policy evaluated, request denied
+	DecisionIgnore = "ignore" // Request not relevant, policy not evaluated
+	DecisionExempt = "exempt" // Request exempt, policy not evaluated
+	DecisionError  = "error"  // Error preventing evaluation, policy not evaluated
+)
+
 type EvaluationRecorder interface {
-	// TODO: fill in args required to record https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#monitoring
-	RecordEvaluation()
+	// TODO: fill in args required to record https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacemenonitoring
+	RecordEvaluation(decision Decision, policy api.LevelVersion, evalMode Mode, attrs api.Attributes)
+}
+
+type PrometheusRecorder struct {
+}
+
+func NewPrometheusRecorder() *PrometheusRecorder {
+	legacyregistry.MustRegister(SecurityEvaluation)
+	return &PrometheusRecorder{}
 }
 
-// TODO: default prometheus-based implementation
+func (r PrometheusRecorder) RecordEvaluation(decision Decision, policy api.LevelVersion, evalMode Mode, attrs api.Attributes) {
+	dec := string(decision)
+	operation := string(attrs.GetOperation())
+	resource := attrs.GetResource().String()
+	subresource := attrs.GetSubresource()
+	if policy.Valid() {
+		v, err := api.ParseVersion(policy.Version.String())
+		var version = v.String()
+		if err != nil || v.Latest() {
+			version = "latest"
+		} else {
+			apiVersion := api.MajorMinorVersion(getAPIVersion())
+			if v.Older(apiVersion) {
```

Comment:
If the versions are the same, we should still record the version (not future).
```suggestion
			if !apiVersion.Older(v) {
```